qlib/kernel/SignalDef.rs

```diff
@@ -85,6 +87,57 @@ impl PtRegs {
         self.rsp = ctx.rsp;
         self.ss = ctx.ss as u64;
     }
+
+    pub fn get_stack_pointer(&self) -> u64 {
+        return self.rsp;
+    }
+
+    pub fn set_stack_pointer(&mut self, sp: u64) {
+        self.rsp = sp;
+    }
+}
+
+#[cfg(target_arch = "aarch64")]
+#[repr(C)]
+#[derive(Debug, Copy, Clone, Default)]
+pub struct PtRegs {
+    pub regs: [u64; 31],
+    pub sp: u64,
+    pub pc: u64,
+    pub pstate: u64,
+    pub orig_x0: u64,
+    pub __pad: u64,
+}
+
+#[cfg(target_arch = "aarch64")]
+impl fmt::LowerHex for PtRegs {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "SP:[{:#x}]
+                   PC:[{:#x}]
+                   State:[{:#x}]
+                   X0:[{:#x}]
+                   GenRegs:[{:#?}]",
+                   self.sp, self.pc, self.pstate,
+                   self.orig_x0, self.regs)
+    }
+}
+#[cfg(target_arch = "aarch64")]
+impl PtRegs {
+    pub fn Set(&mut self, ctx: &SigContext) {
```

Comment:
It seems we can extract a trait defining the behavior of `PtRegs`, but it may not have to be done here.